### ISSUE TITLE
Randomly sample trajectories: Feature request

### ISSUE THREAD

**Issue by Mrugankakarte** [minerllabs/minerl#178]
Hi,

Just a feature request. I was wondering if you could randomly sample trajectories from dataset instead of using all of them in `sarsd_iter` function.

Following is my approach without changing any of your code:
```
def sarsd_iter(self, num_epochs=-1, max_sequence_len=32, queue_size=None, seed=None, include_metadata=False, sample_no = -1):
.
.
.
   
# Setup arguments for the workers.
    if sample_no!=-1 and sample_no > 0:
       files = [(file_dir, max_sequence_len, data_queue, 0, include_metadata) for file_dir in data_list[0:sample_no]]
    files = [(file_dir, max_sequence_len, data_queue, 0, include_metadata) for file_dir in data_list]
.
.
```

Add one parameter for number of trajectories to sample (sample_no).
Since `data_list` has the paths of all trajectories which are shuffled from this line
```
data_list = self._get_all_valid_recordings(self.data_dir)
```
`files` then would read randomly sampled required number of trajectories instead of reading all the files.

Let me know if you can add this!

**Comment by brandonhoughton** [comment]
To summarize - this is looking for a way to limit the number of files sampled by sarsd_iter? 
Something like with a random seed plus % of files (or # of files) you would walk through the files randomly until reaching the desires % or # of files?

**Comment by Mrugankakarte** [comment]
Yes
